src/arrayMethodSort.js

```diff
@@ -5,8 +5,30 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const length = this.length;
+
+    for (let i = 0; i < length - 1; i++) {
+      for (let j = i + 1; j < length; j++) {
+        if (compareFunction && typeof compareFunction === 'function') {
+          if (compareFunction(this[j], this[i]) < 0) {
```

Comment:
second argument is always boolean type, but has to be an element of array, check how sort callback works here 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort